txhelpers/cspp.go

```diff
@@ -62,20 +62,49 @@ func IsMixTx(tx *wire.MsgTx) (isMix bool, mixDenom int64, mixCount uint32) {
 	return
 }
 
-func IsMixedSplitTx(tx *wire.MsgTx, ticketPrice int64) (bool, uint32) {
-	var numTickets uint32
+// FeeForSerializeSize calculates the required fee for a transaction of some
+// arbitrary size given a mempool's relay fee policy.
+func FeeForSerializeSize(relayFeePerKb dcrutil.Amount, txSerializeSize int) dcrutil.Amount {
+	fee := relayFeePerKb * dcrutil.Amount(txSerializeSize) / 1000
+
+	if fee == 0 && relayFeePerKb > 0 {
+		fee = relayFeePerKb
+	}
+
+	if fee < 0 || fee > dcrutil.MaxAmount {
+		fee = dcrutil.MaxAmount
+	}
+
+	return fee
+}
+
+// DefaultRelayFeePerKb is the default minimum relay fee policy for a mempool.
+const DefaultRelayFeePerKb dcrutil.Amount = 1e4
+
+func soloTicketTxSize() int {
+	inSizes := []int{RedeemP2PKHSigScriptSize}
+	outSizes := []int{P2PKHPkScriptSize + 1, TicketCommitmentScriptSize, P2PKHPkScriptSize + 1}
+	return EstimateSerializeSizeFromScriptSizes(inSizes, outSizes, 0)
+}
+
+func IsMixedSplitTx(tx *wire.MsgTx, relayFeeRate, ticketPrice dcrutil.Amount) (isMix bool, numTickets, numMixedIns uint32) {
+	ticketTxFee := FeeForSerializeSize(DefaultRelayFeePerKb, soloTicketTxSize())
+	ticketOutAmt := int64(ticketPrice + ticketTxFee)
+
+	var numOtherOut uint32
 	for _, o := range tx.TxOut {
-		if o.Value == ticketPrice {
+		if o.Value == ticketOutAmt {
```

Comment:
This is all yanked from dcrwallet code.  A way to keep this in agreement with dcrwallet and the CSPP server would be better.